libyul/AssemblyStack.cpp

```diff
@@ -204,6 +204,7 @@ MachineAssemblyObject AssemblyStack::assemble(Machine _machine) const
 		compileEVM(adapter, false, m_optimiserSettings.optimizeStackAllocation);
 		object.bytecode = make_shared<evmasm::LinkerObject>(assembly.assemble());
 		object.assembly = assembly.assemblyString();
+		object.sourceMappings = make_unique<string>(evmasm::AssemblyItem::computeSourceMapping(assembly.items()));
```

Comment:
Maybe it would be better to pass `{scanner().charStream()->name(), 0}` as source indices here and remove the special case for empty source indices in computeSourceMapping.

Reply:
(of course check that `charStream()` is not null and maybe use `""` in that case for the source name).